src/test/java/com/github/jhonnymertz/wkhtmltopdf/wrapper/integration/PdfIntegrationTests.java

```diff
@@ -63,6 +66,25 @@ public void testPdfFromStringTo() throws Exception {
         Assert.assertThat("document should contain the creditorName", pdfText, containsString("Müller"));
     }
 
+    @Test
+    public void testMultipleObjectsWithOptions() throws Exception {
+        final String executable = WrapperConfig.findExecutable();
+        WrapperConfig config = new WrapperConfig(executable);
+        Pdf pdf = new Pdf(config);
+        pdf.addParam( new Param( "--footer-font-size", "10" ) );
+        pdf.addParam( new Param( "--margin-bottom", "20" ) );
+        Cover coverPage = pdf.addCoverFromString("<html><head><meta charset=\"utf-8\"></head><h1>Cover Page</h1></html>");
+        TableOfContents toc = pdf.addToc();
+        toc.addParam(new Param("--disable-dotted-lines"));
+        Page mainPage = pdf.addPageFromString("<html><head><meta charset=\"utf-8\"><style>h2 { page-break-before: always; }</style></head><h2>Heading 1</h2>Blah blah blah<h2>Heading 2</h2>This is some test text</html>");
+        mainPage.addParam(new Param("--footer-center", "Page [page] of [topage]" ));
+        try {
+            pdf.getPDF();
+        } catch (Exception ex) {
+            Assert.fail(ex.getMessage());
+        }
```

Comment:
```suggestion
        byte[] pdfBytes = pdf.getPDF();
        String pdfText = getPdfTextFromBytes(pdfBytes);
        Assert.assertThat("document should contain ...", pdfText, containsString("..."));         
```

No need to try-catch here, but instead add some assertions that validate whether this `pdf.getPDF()` resulted in the desired file.

Reply:
Good suggestion, I will see if i can do something to prove that a particular section of content comes before another. 
And i'll look at adding another test to prove that page specific options are applying and only affect the desired page.